apps/worker/src/monitor/tx.status.monitor.service.ts

```diff
@@ -85,42 +79,47 @@ export class TxStatusMonitoringService extends WorkerHost implements OnApplicati
     // do some stuff
   }
 
-  private async crawlBlockList(txHash: Hash, epoch: string, blockList: bigint[]): Promise<BlockHash | undefined> {
-    const txReceiptPromises: Promise<BlockHash | undefined>[] = blockList.map(async (blockNumber) => {
-      const blockHash = await this.blockchainService.getBlockHash(blockNumber);
-      const block = await this.blockchainService.getBlock(blockHash);
-      const txInfo = block.block.extrinsics.find((extrinsic) => extrinsic.hash.toString() === txHash.toString());
-      this.logger.debug(`Extrinsics: ${block.block.extrinsics[0]}`);
+  private async handleMessagesFailure(jobId: string, moduleError: RegistryError): Promise<void> {
+    this.logger.debug(`Handling extrinsic failure for job ${jobId} and error ${JSON.stringify(moduleError)}`);
+    const txJob = (await this.txReceiptQueue.getJob(jobId)) as Job<ITxMonitorJob, any, string>;
+    if (!txJob) {
+      this.logger.error(`Job ${jobId} not found in tx receipt queue`);
```

Comment:
Do we need to to throw an error or return if the job does not exist or logging it is enough?

Reply:
Good point, I will return error